dspace-api/src/main/java/org/dspace/importer/external/liveimportclient/service/LiveImportClientImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.liveimportclient.service;
+
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.config.RequestConfig.Builder;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@link LiveImportClient}.
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk at 4science dot com)
+ */
+public class LiveImportClientImpl implements LiveImportClient {
+
+    private final static Logger log = LogManager.getLogger();
+
+    private CloseableHttpClient httpClient;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Override
+    public String executeHttpGetRequest(int timeout, String URL, Map<String, String> requestParams) {
```

Comment:
Would there be a way to add tests for this method?  Currently, this PR throws errors because this method isn't working properly.  That implies we need some basic tests of `executeHttpGetRequest()`, as this is a very key method in all these Live Import sources.

Reply:
Moved this feedback over to [#8237](https://github.com/DSpace/DSpace/pull/8237#pullrequestreview-972552327), as that PR is the closest to being merged and this code exists in both these PRs.